pytorch_toolkit/formula_recognition/README.md

```diff
@@ -0,0 +1,180 @@
+
+```bash
+bash init_venv.sh
+```
+
+### Download Datasets
+
+For training model one has to have dataset. Dataset format is similiar to [im2latex-100k](https://zenodo.org/record/56198#.X2NDQ2gzaUl). Main structure of the dataset is following:
+* `formulas_file` - file with one formula per line
+* `images_folder` - folder containing input images
+* `split_file` - this file contains `image_name` (tab symbol) `formula_idx` per line connecting corresponding index of the formula in the file with formulas and particular image with `image_name`. Example:
+    ```
+    11.png  11
+    34.png  34
+    ```
+    There should be at least two such files: `train_filter.lst` and `validate_filter.lst`
+
+> **NOTE**:
+> By default the following structure of the dataset is assumed:
+> `images_processed` - folder with images
+> `formulas.norm.lst` - file with preprocessed formulas, for details, refer to [im2markup](https://github.com/harvardnlp/im2markup) repository
+> `validate_filter.lst` and `train_filter.lst` - corresponding splits of the data.
+
+
+## Training
+
+To train formula recognition model run:
+
+```bash
+python tools/train.py --config configs/config.yml --work_dir <path to work dir>
+```
+Work dir is used to store information about learning: saved model checkpoints, logs.
+
+### Description of possible options in config:
+The config file is divided into 4 sections: train, eval, export, demo. In every section there could exist common parameter as well as unique parameters.
+#### Common parameters:
+ - `backbone_config`:
+    * `arch`: type of the architecture (if backbone_type is resnet). For more details, please, refer to [ResnetLikeBackBone](im2latex/models/backbones/resnet.py)
+    * `disable_layer_3` and `disable_layer_4` - disables layer 3 and 4 in resnet-like backbone
+    * `enable_last_conv` - enables additional convolution layer to adjust number of output channels to the number of input channels in the LSTM. Optional. Default: false.
+    * `in_lstm_ch` - number of input LSTM channels, used only for `last_conv`.
+- `backbone_type`: `resnet` for resnet-like backbone or anything else for original backbone from [im2markup](https://arxiv.org/pdf/1609.04938.pdf) paper. Optional. Default is `resnet`
+- `head` - configuration of the text recognition head. All of the following parameters have default values, you can check them in [text reconition head](im2latex/models/text_recognition_heads/attention_based.py)
+    * `beam_width` - witdth used in beam search. 0 - do not use beam search, 1 and more - use beam search with corresponding number of possible tracks.
+    * `dec_rnn_h` - number of channels in decoding
+    * `emb_size` - dimension of the embedding
+    * `encoder_hidden_size ` - number of channels in encoding
+    * `encoder_input_size ` - number of input in lstm channels, should be equal to `backbone_config.in_lstm_ch`
+    * `max_len` - maximum possible length of the predicted formula
+    * `n_layer` - describe
```

Comment:
what does it describe?

Reply:
Updated readme, it was a mark for me to describe this parameter